environs/__init__.py

```diff
@@ -270,29 +270,70 @@ def read_env(
         file is found. If you do not wish to recurse up the tree, you may pass
         False as a second positional argument.
         """
-        # By default, start search from the same file this function is called
-        if path is None:
+        # declare constants
+        DEFAULT_ENVIRONMENT_FILE_NAME = ".env"
+        PATH_SEPARATOR_PATTERN = r"\\|\/"
+
+        # remember original path
+        original_path = path
+
+        # use .env as path if path is None or empty
+        if not path:
+            path = DEFAULT_ENVIRONMENT_FILE_NAME
+
+        # make path absolute if it is relative, using the directory of the calling script as base
+        if not os.path.isabs(path):
+            def merge_absolute_and_relative_path(absolute_path, relative_path):
```

Comment:
Is this actually necessary? Can you use `pathlib.Path.resolve`?

Reply:
One of your unit tests actually brought me to this function. ;-) When the path contains both a folder and a file name like ""tests/.custom.env", then joining that with the absolute path of the caller leads to a path where "tests" is wrongly included twice, eg. "~/environs/tests/tests/.custom.env". So we need a function that can properly handle situations where folders are included in both the absolute as well as the relative path. The function I wrote was the only option I saw and knew. os.path.join and Path.joinpath do not seem to work here but whenever there is a function that does that out-of-the-box, let's use that function instead.

Reply:
As there is some linting issue left I was not able to solve: could you please replace the "/" against os.path.sep, fix the linting issue and release a new version soon? The code implements the suggested behavior except it explicitly checks if path is a directory or file. Thanks!

Reply:
I also checked pathlib.Path.resolve which did not work.

Reply:
>  The code implements the suggested behavior except it explicitly checks if path is a directory or file

What I'm suggesting [here](https://github.com/sloria/environs/pull/101#issuecomment-531810667) is to only support passing a file and _not_ support passing a directory. 

> fix the linting issue and 

Does `tox -e lint` work for you?

> release a new version soon

This is quite a bit of complexity added to the codebase, so I'd rather not rush this in. I was hoping that supporting only files would keep the implementation simple.

Also, before this is merged, this will need a test that reproduces the issue. Can you add this?

Reply:
Sorry, I was not aware of tox before but finally could get it working. The code throws an exception now when it's given a directory. I also added two tests. I will send a new PR.